jira/resilientsession.py

```diff
@@ -1,94 +1,98 @@
+import abc
 import json
 import logging
 import random
 import time
-from typing import Callable, Optional, Union, cast
+from typing import Optional, Union
 
 from requests import Response, Session
 from requests.exceptions import ConnectionError
+from typing_extensions import TypeGuard
 
 from jira.exceptions import JIRAError
 
-logging.getLogger("jira").addHandler(logging.NullHandler())
+LOG = logging.getLogger(__name__)
 
 
-def raise_on_error(r: Optional[Response], verb="???", **kwargs):
+class PrepareRequestForRetry(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def prepare(self, original_request_kwargs) -> dict:
+        return original_request_kwargs
+
+
+class PassthroughRetryPrepare(PrepareRequestForRetry):
+    def prepare(self, original_request_kwargs) -> dict:
+        return super().prepare(original_request_kwargs)
+
+
+def raise_on_error(resp: Optional[Response], **kwargs) -> TypeGuard[Response]:
     """Handle errors from a Jira Request
 
     Args:
-        r (Optional[Response]): Response from Jira request
-        verb (Optional[str]): Request type, e.g. POST. Defaults to "???".
+        resp (Optional[Response]): Response from Jira request
 
     Raises:
         JIRAError: If Response is None
         JIRAError: for unhandled 400 status codes.
-        JIRAError: for unhandled 200 status codes.
+
+    Returns:
+        bool: True if the passed in Response is all good.
     """
     request = kwargs.get("request", None)
-    # headers = kwargs.get('headers', None)
 
-    if r is None:
+    if resp is None:
         raise JIRAError(None, **kwargs)
 
-    if r.status_code >= 400:
+    if not resp.ok:  # equivalent to .status_code < 400 & >=200
```

Comment:
Great, I'd like to resolve this but I can't find the proper way of doing so, could it be because I didn't have write permission to the repo when the PR was opened? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#resolving-conversations
Anyhow consider this resolved

Reply:
Ahh didn't realise this was a limitation, strange that GitHub allows you to create your own, but not resolve your own !

Thanks for commenting on the threads instead!